geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -2033,6 +2035,29 @@ protected int getRegionSize() {
   }
 
   /**
+   * Returns the size in this region.
+   *
+   * This is used in a transaction to find the size of the region on the transaction hosting node.
+   *
+   * @param target the host of the transaction TXState
+   * @return the number of entries in this region
+   */
+  public int getRegionSize(DistributedMember target) {
```

Comment:
As I know, the LocalRegion APIs are meant to work on local node data store; anything involving messaging remote data store should be in DistributedRegion...
What is the need for moving the code from DistributedTXRegionStub.entryCount() to here?

Reply:
In current transaction implementation, if a region is not partitioned region, it will be treated as a distributed one (see PeerTXStateStub.generateRegionStub). In a sense if both peer nodes have the same LOCAL region name, and involved in a client initiated transaction -- assume one connected to the client is hosting PeerTXStateStub and the other hosting the TXState. A size op of the LOCAL region in the transaction will be forwarded to the node with TXState. You can argue if this is a correct implementation (hopefully a LOCAL region is not involved in a user transaction), however, I do not want to change the current behavior of the transaction. If needed, we can open a new ticket whether we should throw exception when a local transaction is involved in a transaction. Otherwise, I think current behavior (all operations in a transaction is executed on the node hosting TXState is acceptable.)